doc/decap/subnet_decap_HLD.md

```diff
@@ -0,0 +1,194 @@
+| decap TTL mode  | pipe                          |                                  |
+| decap DSCP mode | uniform                       |                                  |
+
+The decapsulation termination entry will be created with the following attributes:
+| Attribute       | Value                               | Note                                                                                                                     |
+| --------------- | ----------------------------------- | ------------------------------------------------------------------------------------------------------------------------ |
+| term entry type | MP2MP                               | multi-point to multi-point                                                                                               |
+| dest IP         | the VLAN subnet                     |                                                                                                                          |
+| dest IP mask    | the vlan subnet mask                |                                                                                                                          |
+| source IP       | Netscan privately-owned subnet      | IPinIP packets that have source IP in the subnet can be safely assumed to be Netscan traffic instead of customer traffic |
+| source IP mask  | Netscan privately-owned subnet mask |                                                                                                                          |
+
+### 6.2 DB Schema
+
+#### 6.2.1 CONFIG_DB
+
+```
+### SUBNET_DECAP
+    ; Stores subnet based decapsulation configurations
+    key                 = SUBNET_DECAP|subnet_type                      ; SUBNET_DECAP|vlan
+    status              = "enable"/"disable"                            ; status of subnet based decapsulation
+    src_ip              = source IP prefix                              ; source IP prefix used for decap terms of IPv4 tunnel
+    src_ip_v6           = source IP prefix                              ; source IPv6 prefix used for decap terms of IPv6 tunnel
+```
+
+#### 6.2.2 APPL_DB
+
+```
+### TUNNEL_DECAP_TABLE
+    ; Stores a list of decap tunnels
+    key                 = TUNNEL_DECAP_TABLE:tunnel_name                ; tunnel name as key
+    tunnel_type         = "IPINIP"                                      ; tunnel type
+    dscp_mode           = "uniform"/"pipe"
+    ecn_mode            = "copy_from_outer"/"standard"
+    ttl_mode            = "uniform"/"pipe"
+    encap_ecn_mode      = "standard"
+
+### TUNNEL_DECAP_TERM_TABLE
+    ; Stores a list of decap terms.
+    key                 = TUNNEL_DECAP_TERM_TABLE:tunnel_name:dst_ip    ; tunnel name:dst IP prefix as key
+    term_type           = "P2P"/"P2MP"/"MP2MP"                          ; tunnel decap term type
+    src_ip              = source IP prefix
+```
+
+#### 6.2.3 STATE_DB
+
+```
+### TUNNEL_DECAP_TABLE
```

Comment:
why add new state db? Its role is not indicated in the following workflow diagram

Reply:
updated the workflow diagram to indicated the state db.